pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -615,6 +616,39 @@ impl RemoteTimelineClient {
         Ok(())
     }
 
+    // NOTE: if there were no tasks to call stop we need to call stop by ourselves first
+    pub(crate) async fn persist_index_part_with_deleted_flag(
+        self: &Arc<Self>,
+    ) -> anyhow::Result<()> {
+        self.stop().context("stop")?;
```

Comment:
I think the way currently stop is working is suboptimal, but decided not to touch it here. IMO ideally we should have non cooperative stop, when hypothetical Timeline::stop calls into RemoteTimelineClient::stop, which joins all tasks and shuts down after that.